ipatests/test_integration/test_caless.py

```diff
@@ -1181,6 +1181,36 @@ def test_no_ds_password(self):
         if self.domain_level > DOMAIN_LEVEL_0:
             self.verify_installation()
 
+    @replica_install_teardown
+    def test_install_ca_replica(self):
+        # related to https://pagure.io/freeipa/issue/6985
+        replica = self.replicas[0]
+
+        self.create_pkcs12('ca1/replica', filename='server.p12')
+        result = self.prepare_replica(http_pkcs12='server.p12',
+                                      dirsrv_pkcs12='server.p12')
+        assert result.returncode == 0
+        if self.domain_level > DOMAIN_LEVEL_0:
+            self.verify_installation()
+
+        contents = (
+            '-----BEGIN CERTIFICATE-----\n'
+            'sdnmsdkfbsdifbsdbasdsdSDDDasdmnd\n'
+            '-----END CERTIFICATE-----')
+        cert1 = tempfile.mkdtemp(suffix='xyz.crt', dir=paths.TMP)
+        replica.put_file_contents(cert1, contents)
+        cert_files = [cert1]
+        # install ca directly with option --external-cert-file
+        result = tasks.install_ca(replica, cert_files=cert_files,
+                                  raiseonerr=False)
+        assert_error(result, 'First run the installer with --external-ca')
+
+        # install ca with invalid cert
+        tasks.install_ca(self.master, external_ca=True)
+        result = tasks.install_ca(replica, cert_files=cert_files,
+                                  raiseonerr=False)
+        assert_error(result, 'CA is not installed yet')
+
```

Comment:
Hi @amore17 
thank you for the test. If I understand correctly, it is testing the following scenario:

- install master CA-less
- install replica
- call ipa-ca-install --external-cert-file on the replica (without the first step ipa-ca-install --external-ca)
It is a valid test case, but I think that the bug 6985 was rather related to the following scenario:
- install master with CA
- install replica
- run ipa-ca-install --external-cert-file on the replica
In this case the replica should complain that CA instance is already configured on the master and external-cert-file will be ignored.

Reply:
Thanks @flo-renaud for review I'll add that scenario in ipatests/test_integration/test_commands.py
should I remove this test-case ?

Reply:
You can keep the test, it's a valid test case.